### DISCUSSION TITLE
Type alias rendering

### DISCUSSION THREAD

**Discussion by AntoineD** [mkdocstrings/python#137] [Q&A]
Is there an option to display the definition of a type alias instead of its name?
For instance, to render `Alias = str | None` as `str | None` instead of `Alias` where used in a signature.

**Comment by pawamoy** [comment]
There's no such option yet. The recommended way is to document your type alias so that user can click on it and see its value, itself cross-referencing other type aliases it uses, etc.

I'm not sure how to detect that `Alias = str | None` is a type alias and that it should be expanded when rendering it. Also, if you have nested aliases, should the expansion be recursive? Signatures could become difficult to read with complex types if we recursively expand all type aliases.

Previous related discussion, for literals: https://github.com/mkdocstrings/python/issues/73.

**Reply by AntoineD** [reply]
Too bad this cannot be chosen.

The default behavior in sphinx is to show the type alias definition, with an [option](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_type_aliases) to do otherwise.

**Reply by pawamoy** [reply]
Thanks for the link.

I thought about it and I think we could do this: add an option to tell mkdocstrings-pyton to recursively expand all types used in signatures. If a name in an expression points to an attribute, this attribute's value will be used to expand the name. Names in this value will do the same, etc.

**Reply by AntoineD** [reply]
That would be awesome!

**Reply by AntoineD** [reply]
Hello,
another argument in favor of such an option came to my mind: with nested type aliases, end users would have to click around to get the definitions of all the type parts, which I think is not ideal.

**Reply by pawamoy** [reply]
Coming back to this: @AntoineD could you open a feature request? You can just link to [this comment](https://github.com/mkdocstrings/python/discussions/137#discussioncomment-8646191). Hard to keep track of things in discussions :slightly_smiling_face:
